pkg/report/detailer.go

```diff
@@ -0,0 +1,77 @@
+/*
+ * @license
+ * Copyright 2024 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package report
+
+import (
+	"context"
+	"fmt"
+)
+
+const (
+	TypeInfo  string = "INFO"
+	TypeWarn  string = "WARN"
+	TypeError string = "ERROR"
+)
+
+type Detail struct {
+	Type    string `json:"type"`
+	Message string `json:"msg"`
+}
+
+type detailerContextKey struct{}
+
+func NewContextWithDetailer(ctx context.Context, d Detailer) context.Context {
```

Comment:
Is it more `AddDetailerToContext`

Reply:
Would also work, but I took this name because similar functions are already in use, e.g. `logr.NewContext(...)` or `newContextWithEnvironment(...)`